config_src/drivers/nuopc_cap/mom_cap.F90

```diff
@@ -489,7 +486,33 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
 
 !$  call omp_set_num_threads(nthrds)
 
-  call fms_init(mpi_comm_mom)
+  ! reset shr logging to my log file
+  if (localPet==0) then
+    call NUOPC_CompAttributeGet(gcomp, name="diro", &
+         isPresent=isPresentDiro, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    call NUOPC_CompAttributeGet(gcomp, name="logfile", &
+         isPresent=isPresentLogfile, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    if (isPresentDiro .and. isPresentLogfile) then
+         call NUOPC_CompAttributeGet(gcomp, name="diro", value=diro, rc=rc)
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+         call NUOPC_CompAttributeGet(gcomp, name="logfile", value=logfile, rc=rc)
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+         open(newunit=stdout,file=trim(diro)//"/"//trim(logfile))
```

Comment:
@jedwards4b , here the `newunit` argument returns a negative number `-10`. Within FMS, unit numbers are used as array indices (see mpp_io_misc.inc), so when the unit is -10, i get the following error:

```
Fortran runtime error: Index '-10' of dimension 1 of array 'mpp_file' below lower bound of -1
```

You need to run a GNU debug test to reproduce the error.

Reply:
Yes newunit returns a negative value - did you try with the latest FMS_Interface?   I said it wasn't required, but perhaps it is.

Reply:
Yes, I tried it with the latest FMS_interface and 2021.03.01